subsystems/mvdm/ntvdm/bios/vidbios.c

```diff
@@ -2942,6 +2942,295 @@ static VOID VidBiosPrintCharacter(CHAR Character, BYTE Attribute, BOOLEAN UseAtt
+            OldRead = IOReadB(VGA_GC_DATA);
+
+            /* Save old mask */
+            IOWriteB(VGA_SEQ_INDEX, VGA_SEQ_MASK_REG);
+            OldMask = IOReadB(VGA_SEQ_DATA);
+
+            /* Loop and write over all four color bitplanes */
+            CurrentPlaneMask = 1;
+            for (CurrentPlane = 0; CurrentPlane < 4; CurrentPlane++)
+            {
+                BYTE Dummy;
+                BYTE ActualValue;
+
+                IOWriteB(VGA_GC_INDEX, VGA_GC_READ_MAP_SEL_REG);
+                IOWriteB(VGA_GC_DATA, CurrentPlane);
+                IOWriteB(VGA_SEQ_DATA, CurrentPlaneMask);
+                EmulatorReadMemory(&EmulatorContext, Address, &Dummy, sizeof(BYTE));
+
+                /* There's probably a way to handle this logic on write, but I'll just leave it at that */
+                if (Xor)
+                {
+                    ActualValue = (Value & 1) << Shift;
+                }
+                else
+                {
+                    ActualValue = Dummy;
+                    ActualValue &= ~(1 << Shift);
+                    ActualValue |= (Value & 1) << Shift;
+                }
+
+                EmulatorWriteMemory(&EmulatorContext, Address, &ActualValue, sizeof(BYTE));
+
+                Value >>= 1;
+                CurrentPlaneMask <<= 1;
+            }
+
+            /* Restore old registers */
+            IOWriteB(VGA_GC_INDEX, VGA_GC_MODE_REG);
+            IOWriteB(VGA_GC_DATA, OldMode);
+            IOWriteB(VGA_GC_INDEX, VGA_GC_ROTATE_REG);
+            IOWriteB(VGA_GC_DATA, OldRotation);
+            IOWriteB(VGA_GC_INDEX, VGA_GC_READ_MAP_SEL_REG);
+            IOWriteB(VGA_GC_DATA, OldRead);
+            IOWriteB(VGA_SEQ_INDEX, VGA_SEQ_MASK_REG);
+            IOWriteB(VGA_SEQ_DATA, OldMask);
+            break;
+        }
+
+        case 0x11: /* 640x480 2c */
+        {
+            ULONG Address;
+            BYTE MemoryValue;
+            BOOL Xor;
+            BYTE Shift = 7 - X % 8;
+
+            Xor = Value & 0x80;
+            Value &= 1;
+            Address = TO_LINEAR(GRAPHICS_VIDEO_SEG,
+                                Page * Bda->VideoPageSize + (Y * ScreenWidth + X) / 8);
+
+            EmulatorReadMemory(&EmulatorContext, Address, &MemoryValue, sizeof(BYTE));
+            if (Xor)
+            {
+                MemoryValue ^= Value << Shift;
+            }
+            else
+            {
+                MemoryValue &= ~(1 << Shift);
+                MemoryValue |= Value << Shift;
+            }
+            EmulatorWriteMemory(&EmulatorContext, Address, &MemoryValue, sizeof(BYTE));
+            break;
+        }
+
+        case 0x13: /* 320x200 256c */
+        {
+            WORD Offset = Y * ScreenWidth + X;
+            EmulatorWriteMemory(
+                &EmulatorContext,
+                TO_LINEAR(GRAPHICS_VIDEO_SEG, Page * Bda->VideoPageSize + Offset),
+                (PVOID)&Value,
+                sizeof(BYTE));
```

Comment:
```suggestion
            EmulatorWriteMemory(&EmulatorContext,
                                TO_LINEAR(GRAPHICS_VIDEO_SEG,
                                          Page * Bda->VideoPageSize + (Y * ScreenWidth + X)),
                                (PVOID)&Value,
                                sizeof(BYTE));
```
Also I bet the compiler doesn't complain if you remove the PVOID cast.

Reply:
It doesn't complain, I just forgot to remove the cast

Reply:
No problem; I know there are other places with the cast where it's not necessary either.